### PR TITLE
BUG: Line plots aligned from start #57594

### PR DESCRIPTION
**Pull request by NoelMT**
#57594 BUG: df.plot makes a shift to the right if frequency multiplies to n > 1

The cause of this bug is that by default all elements are generated to align to the end of
a period instead of at the start and the period technically does not end at the start of
the last period but rather just before the period subsequent to the last period. To fix this
problem we therefore passed a parameter stating that the elements should be aligned to
the start of the period instead of the end. Parameter "how=S" was added in pandas/plotting/_matplotlib/timeseries.py::maybe_convert_index(), 
on the function call data.index.asfreq(freq=freq_str, how = "S").  

Additional tests were added to test whether input data match data in plot and one test was merged into one the added test to reduce code duplication.

### CODE REVIEW

**Comment by datapythonista** [comment]
Thank you for working on this @NoelMT. Do you mind adding plots to this PR (as a comment or in the description) that show how things are rendered before and after your changes please? That would give context to reviewers and it'll make reviewing much easier. Thanks!

File: pandas/plotting/_matplotlib/timeseries.py [hunk 1]
```diff
@@ -310,7 +310,7 @@ def maybe_convert_index(ax: Axes, data: NDFrameT) -> NDFrameT:
             if isinstance(data.index, ABCDatetimeIndex):
                 data = data.tz_localize(None).to_period(freq=freq_str)
             elif isinstance(data.index, ABCPeriodIndex):
-                data.index = data.index.asfreq(freq=freq_str)
```
Comment (natmokval) [inline pandas/plotting/_matplotlib/timeseries.py:313]: Could you please check `ruff`, `ruff-format` and `codespell` errors on `line 313`?
If you try following command: `pre-commit run --all-file`, it should help fix these errors.

File: pandas/plotting/_matplotlib/timeseries.py [hunk 2]
```diff
@@ -310,7 +310,7 @@ def maybe_convert_index(ax: Axes, data: NDFrameT) -> NDFrameT:
...
+                data.index = data.index.asfreq(freq=freq_str, how = "S") #Lineplot alignes to the start of period
```
Comment (natmokval) [inline pandas/plotting/_matplotlib/timeseries.py:313]: Could you please check `ruff`, `ruff-format` and `codespell` errors on `line 313`?
If you try following command: `pre-commit run --all-file`, it should help fix these errors.

File: pandas/tests/indexes/period/test_period_range.py [hunk 3]
```diff
@@ -113,34 +119,6 @@ def test_construction_from_string(self, freq_offset, freq_period):
         result = period_range(start=end, end=start, freq=freq_period, name="foo")
         tm.assert_index_equal(result, expected)
 
-    def test_construction_from_string_monthly(self):
```
Comment (natmokval) [inline pandas/tests/indexes/period/test_period_range.py:116]: Thank you for noticing code duplication in `pandas/tests/indexes/period/test_period_range.py`. I am not sure how it’s related to this bug.  Could you please move these changes in separate PR?

File: pandas/tests/plotting/test_datetimelike.py [hunk 4]
```diff
@@ -69,6 +70,31 @@ def test_fontsize_set_correctly(self):
         for label in ax.get_xticklabels() + ax.get_yticklabels():
             assert label.get_fontsize() == 2
 
+    @pytest.mark.parametrize("freq", ["20s","3s","60min","5min","7h","4D","8W","11M","2Q","Y", "3Y", "10Y"])
```
Comment (natmokval) [inline pandas/tests/plotting/test_datetimelike.py:73]: `pre-commit run --all-file` fails. Could you please fix it?

File: pandas/tests/plotting/test_datetimelike.py [hunk 5]
```diff
@@ -69,6 +70,31 @@ def test_fontsize_set_correctly(self):
...
+    def test_period_range_content(self, freq):
+        # Setup your DataFrame as in your scenario
```
Comment (natmokval) [inline pandas/tests/plotting/test_datetimelike.py:75]: instead of the comments, e.g. `# Setup your DataFrame as in your scenario`, could you please just add the number of GitHub issue: `# GH#57594`

**Comment by natmokval** [comment]
Thanks @NoelMT for working on this. Seems like it will fix the problem with visualisation of `DataFrame` if `PeriodIndex` as index provided.
e.g. for the first scenario:
```
from pandas import *
import numpy as np

idx = period_range("01/01/2000", freq='7h', periods=4)
df = DataFrame(
    np.array([0, 1, 0, 1]),
    index=idx,
    columns=["A"],
)
df.plot()
print(df)
```

plot before:

res_7h

plot after:

dfplot_7h_correct

**Comment by natmokval** [comment]
@NoelMT could you please add a note to `doc/source/whatsnew/v3.0.0.rst` in the section
https://github.com/pandas-dev/pandas/blob/dc19148bf7197a928a129b1d1679b1445a7ea7c7/doc/source/whatsnew/v3.0.0.rst?plain=1#L356

**Comment by mroeschke** [comment]
Thanks for the pull request, but it appears to have gone stale. If interested in continuing, please merge in the main branch, address any review comments and/or failing tests, and we can reopen.
